Sprint-1/destructuring/exercise-3/exercise.js

```diff
@@ -6,3 +6,17 @@ let order = [
   { itemName: "Hot Coffee", quantity: 2, unitPricePence: 100 },
   { itemName: "Hash Brown", quantity: 4, unitPricePence: 40 },
 ];
+
+function printOrder(order){
+  console.log(`${'QTY'.padEnd(8)}${'ITEM'.padEnd(20)}TOTAL`);
+
+  let total = 0;
+  order.forEach(({itemName, quantity, unitPricePence}) => {
+    const itemTotalPrice = unitPricePence / 100 * quantity;
+    total += itemTotalPrice;
+    console.log(`${quantity.toString().padEnd(8)}${itemName.padEnd(20)}${itemTotalPrice.toFixed(2)}`);
+  });
```

Comment:
You are applying correct approach when you need to iterate over the whole array and to count total value. But, there is a better option for the same task. Again, this doesn't mean that your code is not valid, but do you know what is the other way to achieve the same goal?

Reply:
function printOrder(order) {
  console.log(`${'QTY'.padEnd(8)}${'ITEM'.padEnd(20)}TOTAL`);

  const total = order.reduce((runningTotal, { itemName, quantity, unitPricePence }) => {
    const itemTotalPrice = unitPricePence / 100 * quantity;
    runningTotal += itemTotalPrice;
    console.log(`${quantity.toString().padEnd(8)}${itemName.padEnd(20)}${itemTotalPrice.toFixed(2)}`);
    return runningTotal;
  }, 0);

  console.log(`\nTotal: ${total.toFixed(2)}`);
}

I can see, there is another solution with the reduce() method. So, the reduce() will do 2 steps in one: loop through the array and calculate the total. It's a clearer and easier way to handle this, thank you for asking me this question. It made me think in another way.